server-spi/src/main/java/org/keycloak/storage/user/UserQueryProvider.java

```diff
@@ -237,35 +240,53 @@ default int getUsersCount(RealmModel realm, boolean includeServiceAccount) {
     }
 
     /**
-     * Search for user by parameter.  Valid parameters are:
-     * "first" - first name
-     * "last" - last name
-     * "email" - email
-     * "username" - username
+     * Search for user by a map of parameters.
+     * <p/>
+     * Valid parameters are:
+     * <ul>
+     *     <li>"firstName" - first name (case insensitive string)</li>
```

Comment:
Prefer constants, e.g. UserModel.FIRST_NAME to plain strings.
```suggestion
     *     <li>UserModel.FIRST_NAME - first name (case insensitive string)</li>
```

Same comment applies to methods below